File: swing/src/main/java/wannabe/swing/renderer/Square.java
package wannabe.swing.renderer;

import java.awt.Graphics;
import wannabe.swing.SwingProjected;

public class Square extends SwingRenderer {
  @Override public void draw(Graphics g, SwingProjected p) {
    g.drawRect(p.left, p.top, p.size, p.size);
  }
}

File: swing/src/main/java/wannabe/swing/renderer/Square.kt
package wannabe.swing.renderer

import wannabe.swing.SwingProjected
import java.awt.Graphics

class Square : SwingRenderer() {
  override fun draw(
    g: Graphics,
    p: SwingProjected
  ) {
    g.drawRect(p.left, p.top, p.size, p.size)
  }
}
